### PR TITLE
for #5: start browser action UI

### PR DESCRIPTION
**Pull request by groovecoder**
testpilot-containers-screenshot

Still needs:
* Working sort button (See #9)
* [x] Container icons
* [x] Working Edit button

### CODE REVIEW

**Comment by groovecoder** [comment]
@jonathanKingston - neither the `tabs` WebExtenion API, nor the SDK API have any kind of `.contextualIdentity` property, so I can't write a sort directly on those `tab`s objects. Is it something we could add to the `browser.contextualidentity` API? E.g., add a `tabs` property to the `identity` object:
```
{
    name: ContextualIdentityService.getUserContextLabel(identity.userContextId),
    icon: identity.icon,
    color: identity.color,
    cookieStoreId: getCookieStoreIdForContainer(identity.userContextId),
    tabs: [<tab-ids-for-the-container>]
}
```
?

**Comment by jonathanKingston** [comment]
@groovecoder you should be able to check the cookieStoreID of the tabs right? Giving the same API functionality.

**Comment by groovecoder** [comment]
So this is working like so ...
#### Install the webextension-experiment to get the `browser.contextualIdentity` API
1. `cd webextension-experiment && zip contextualIdentities.xpi api.js install.rdf schema.json`
2. Go to `about:addons`
3. Click the gear button, and "Install Add-on From File"
4. Select the `contextualIdentities.xpi` file
#### Install the webextension (or use jpm)
1. Go to `about:debugging`
2. Click "Load Temporary Add-on"
3. Select the `webextension/manifest.json`

Still TODO:
* [ ] Custom containers show the wrong icon & color.

**Comment by groovecoder** [comment]
Added a WIP commit for fixing custom container styles, so don't merge this PR!!

**Comment by groovecoder** [comment]
Updated to fix icons & colors for all containers in the pop-up.

**Comment by groovecoder** [comment]
Since we already have #9 for the container sort button, I'd like to call this PR done - it implements everything else described in #5. @jonathanKingston - can you review & spot-check and merge?

**Comment by groovecoder** [comment]
@jonathanKingston pointed out that the color **names** don't match the color **values** of containers. So we need to restore his map of color **names** to color **values** in the `popup.css`.

File: webextension/css/popup.css [hunk 1]
```diff
@@ -51,3 +51,63 @@ table.unstriped tbody tr {
...
+  --identity-icon-color: #ee5195;
+}
+
+[data-identity-color="purple"] {
+  --identity-tab-color: #7a2f7a;
+  --identity-icon-color: #7a2f7a;
+}
+
+[data-identity-icon="fingerprint"] {
+  --identity-icon: url("chrome://browser/content/usercontext.svg#fingerprint");
```
Comment (groovecoder) [inline webextension/css/popup.css]: Oh cool, I didn't think `chrome://` resources would be available in a WebExtension popup.

File: webextension/js/popup.js [hunk 2]
```diff
@@ -5,29 +5,16 @@ browser.contextualIdentities.query({}).then(identites=> {
...
   });
-
-  const customContainerStyleElement = document.createElement('style');
-
-  customContainerStyleElement.type = 'text/css';
-  customContainerStyleElement.appendChild(document.createTextNode(customContainerStyles));
-
-  const head = document.head;
-
-  head.appendChild(customContainerStyleElement);
```
Comment (groovecoder) [inline webextension/js/popup.js]: Yay less code!

File: webextension/css/popup.css (hunk 1 above)
Comment (jonathanKingston) [inline webextension/css/popup.css]: @groovecoder I checked with dholbert they are for backgrounds but not for filters.

**Pull request merged**